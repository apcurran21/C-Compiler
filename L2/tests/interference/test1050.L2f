(@main
	0
	:call_label_ciao__ciao_13
	goto :call_label_ciao__ciao_18
	:call_label_ciao__ciao_14
	%pr <- %searchCounter
	%pr <<= 1
	%pr += 1
	rdi <- %pr
	call print 1
	return
	:call_label_ciao__ciao_15
	%primeCounter += 1
	goto :call_label_ciao__ciao_16
	:call_label_ciao__ciao_16
	%weDone <- 58 <= %primeCounter
	cjump %weDone = 1 :call_label_ciao__ciao_14
	goto :call_label_ciao__ciao_17
	:call_label_ciao__ciao_17
	%searchCounter += 1
	rdi <- %searchCounter
	mem rsp -8 <- :call_label_ciao__ciao_19
	call @prime 1
	:call_label_ciao__ciao_19
	%result <- rax
	%nresult <- %result = 0
	cjump %nresult = 1 :call_label_ciao__ciao_16
	goto :call_label_ciao__ciao_15
	:call_label_ciao__ciao_18
	%primeCounter <- 0
	%searchCounter <- 0
	goto :call_label_ciao__ciao_16
)
