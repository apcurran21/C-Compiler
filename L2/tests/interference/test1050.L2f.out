%nresult %primeCounter r12 r13 r14 r15 rbp rbx %searchCounter
%primeCounter %nresult r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %result rsi %searchCounter %weDone
%pr r12 r13 r14 r15 rbp rbx
r10 %primeCounter r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %searchCounter
r11 %primeCounter r10 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %searchCounter
r12 %nresult %pr %primeCounter r10 r11 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %result rsi %searchCounter %weDone
r13 %nresult %pr %primeCounter r10 r11 r12 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %result rsi %searchCounter %weDone
r14 %nresult %pr %primeCounter r10 r11 r12 r13 r15 r8 r9 rax rbp rbx rcx rdi rdx %result rsi %searchCounter %weDone
r15 %nresult %pr %primeCounter r10 r11 r12 r13 r14 r8 r9 rax rbp rbx rcx rdi rdx %result rsi %searchCounter %weDone
r8 %primeCounter r10 r11 r12 r13 r14 r15 r9 rax rbp rbx rcx rdi rdx rsi %searchCounter
r9 %primeCounter r10 r11 r12 r13 r14 r15 r8 rax rbp rbx rcx rdi rdx rsi %searchCounter
rax %primeCounter r10 r11 r12 r13 r14 r15 r8 r9 rbp rbx rcx rdi rdx rsi %searchCounter
rbp %nresult %pr %primeCounter r10 r11 r12 r13 r14 r15 r8 r9 rax rbx rcx rdi rdx %result rsi %searchCounter %weDone
rbx %nresult %pr %primeCounter r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rcx rdi rdx %result rsi %searchCounter %weDone
rcx %primeCounter r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rdi rdx rsi %searchCounter
rdi %primeCounter r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdx rsi %searchCounter
rdx %primeCounter r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rsi %searchCounter
%result %primeCounter r12 r13 r14 r15 rbp rbx %searchCounter
rsi %primeCounter r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %searchCounter
%searchCounter %nresult %primeCounter r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %result rsi %weDone
%weDone %primeCounter r12 r13 r14 r15 rbp rbx %searchCounter
